index.js

```diff
@@ -30,16 +30,34 @@ module.exports = function generateRequireForUserCode (scopedDirs, options) {
     }
   })
 
-  function deleteModuleFromCache (m) {
+  function isSubPath (parent, modulePath) {
+    const relative = path.relative(parent, modulePath)
+    return relative && !relative.startsWith('..')
+  }
+
+  function shouldDeleteFromCache (modulePath, directoriesToClear) {
+    if (_.isEmpty(directoriesToClear) || modulePath === baseModule.id) {
```

Comment:
```js
_.isEmpty(directoriesToClear)
```
means it will clear all folders if `[]` was explicitly passed as `directoriesToClear` - isn't this a bit confusing?

Reply:
Actually yeah, very good point(though really no reason to call `clearCache([])` 🤣  )
Changed to `=== undefined` here: https://github.com/wix/scoped-require/pull/11/commits/3482c122797da28a99add83d954e0a1bfd3562b0